File: MK/10월 2주차/구명보트.java
/**
 * 구명보트
 * 문제: https://school.programmers.co.kr/learn/courses/30/lessons/42885
 */

import java.util.*;

class Solution {
    public int solution(int[] people, int limit) {
        int res=0;
        int n=people.length;

        Arrays.sort(people);

        int left=0;
        int right=n-1;

        while (left < right) {
            if (people[left] + people[right] <= limit) {
                res++;
                left++;
                right--;
            } else {
                right--;
            }
        }

        return n-res;
    }
}

File: BW/10월3주차/구명보트.js
function solution(people, limit) {
    var answer = 0;
    var arr = people;
    var start = 0;
    var end = arr.length-1;
    arr.sort((a, b) => a - b); 
    
    while(1){
        if(start === end) break;
        if(arr[start]+arr[end] <= limit) start++;
        end++;
        answer++;
    }
    
    return answer;
}